Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClientCache.cs

```diff
@@ -0,0 +1,146 @@
+            StartBackgroundCleanup();
+        }
+
+        internal void StartBackgroundCleanup()
+        {
+            lock (_backgroundCleanupLock)
+            {
+                if (_backgroundCleaningTask != null)
+                {
+                    _backgroundCleaningTaskTokenSource.Cancel();
+                    try
+                    {
+                        _backgroundCleaningTask.Wait();
+                    }
+#pragma warning disable ERP022 // Unobserved exception in generic exception handler
+                    catch (AggregateException) { }
+#pragma warning restore ERP022  // Unobserved exception in generic exception handler
+                }
+
+                _backgroundCleaningTaskTokenSource = new CancellationTokenSource();
+                _backgroundCleaningTask = Task.Run(() => BackgroundCleanupAsync());
+            }
+        }
+
+        private async Task BackgroundCleanupAsync()
+        {
+            var ct = _backgroundCleaningTaskTokenSource.Token;
+
+            while (!ct.IsCancellationRequested)
+            {
+                var earliestLastUseTime = DateTime.UtcNow - TimeSpan.FromMinutes(_maximumAgeInMinutes);
+
+                var shutdownTasks = new List<Task<BoolResult>>();
+                foreach (var kvp in _clientDict)
+                {
+                    var client = kvp.Value;
+                    if (client._uses == 0 && client._lastUseTime < earliestLastUseTime)
+                    {
+                        lock (_backgroundCleanupLock)
+                        {
+                            if (client._uses == 0
+                                && client._lastUseTime < earliestLastUseTime
+                                && _clientDict.TryRemove(client.Key, out GrpcCopyClient removedClient))
+                            {
+                                // Cannot await within a lock
+                                shutdownTasks.Add(client.ShutdownAsync(_context));
+                            }
+                        }
+                    }
+                }
+
+                var allTasks = Task.WhenAll(shutdownTasks.ToArray());
+                try
+                {
+                    await allTasks;
+                }
+#pragma warning disable ERP022 // Unobserved exception in generic exception handler
+                catch (Exception)
+                {
+                    // If Task.WhenAll throws in an await, it unwraps the AggregateException and only
+                    // throws the first inner exception. We want to see all failed shutdowns.
+                    _context.Error($"Shutdown of unused GRPC clients failed after removal from client cache. {allTasks.Exception}");
+                }
+#pragma warning restore ERP022 // Unobserved exception in generic exception handler
+
+                await Task.Delay(TimeSpan.FromMinutes(_waitBetweenCleanupInMinutes), ct);

```

Comment:
I think during cancellation we can get UnobservedTaskException, because this Delay will fail with TaskCancelled, and I can't see that _backgroundCleaningTask exception is handled...